plugins/optimization-detective/load.php

```diff
@@ -25,6 +25,24 @@
 	return;
 }
 
+if (
+	( is_admin() || ( defined( 'WP_CLI' ) && WP_CLI ) ) &&
+	! file_exists( __DIR__ . '/detection/web-vitals.asset.php' )
+) {
+	// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
+	trigger_error(
```

Comment:
Why not use `wp_trigger_error()`? Or even more obvious with an admin notice, like for instance in https://github.com/ampproject/amp-wp/blob/develop/amp.php#L178?

Reply:
The problem with `wp_trigger_error()` is that it does nothing if `WP_DEBUG` isn't enabled (which seems somewhat strange especially if `E_USER_ERROR` had been passed).

As for a more robust error notices like in the AMP plugin, I think that might be overkill for Optimization Detective. There are a lot more dependencies for the AMP plugin which can cause errors to happen a lot more often, so more users will be seeing that notice, even end users. For Optimization Detective, the only users seeing the error would be developers cloning from GitHub. So I think what we have now is sufficient.

Reply:
Makes sense regarding `wp_trigger_error()`. Or rather not - well...

Regarding the AMP plugin, as to what I had mentioned before, there's also https://github.com/ampproject/amp-wp/blob/develop/amp.php#L162 which shows an admin notice for when the plugin isn't built. That's just as much applicable here, which is why I was proposing it.

But not a necessity to have either way.

Reply:
Yeah, for that we just included it among the other load error checks since the infrastructure was already there. But since that is a build time error rather than a runtime error, it technically isn't needed for the production build.